### ISSUE TITLE
Feature Request: do not wrap SqlExceptions if dao interface method declares it

### ISSUE THREAD

**Issue by radai-rosenblatt** [jdbi/jdbi#148]
if i have a dao interface that explicitely declares an SqlException as being thrown out of a method

```java
public interface myDao {
   @SqlUpdate("insert into table values ...")
   int insert(MyClass instance) throws SqlException;
}
```

would it be possible not to wrap SqlExceptions coming out of the generated code with UnableToExecuteStatementException ?

as it is right now i cannot write (in some higher order logic class):

```java
try {
   myDao.insert(something);
} catch (SqlException e) {
   //handle known problems
}
```

but i must catch UnableToExecuteStatementException, which is jdbi-internal

**Comment by stevenschlansker** [comment]
I actually kinda like this, seems unlikely to make JDBI2, but there's been talk of having a pluggable "exception policy"

**Comment by qualidafial** [comment]
This was resolved in #312
